sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultCredentialPolicy.java

```diff
@@ -340,4 +358,24 @@ private static String getRequestAuthority(HttpRequest request) {
 
         return authority;
     }
+
+    private static boolean isChallengeResourceValid(HttpRequest request, String scope) {
+        final URI scopeUri;
+
+        try {
+            scopeUri = new URI(scope);
+        } catch (URISyntaxException e) {
+            // The challenge contains an invalid scope.
+            return false;
+        }
+
+        if (!request.getUrl().getHost().toLowerCase(Locale.ROOT)
```

Comment:
Is this going to be called for each request made? If so you should look at using [`String.regionMatches`](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#regionMatches-boolean-int-java.lang.String-int-int-) which should provide better performance that lowercasing both strings.

Reply:
It would be called only for the first "naive" request we make to get an authentication challenge from the service and whenever the token we generate from that process expires.